Modules/_io/winconsoleio.c

```diff
@@ -610,7 +610,7 @@ static wchar_t *
 read_console_w(HANDLE handle, DWORD maxlen, DWORD *readlen) {
     int err = 0, sig = 0;
 
-    wchar_t *buf = (wchar_t*)PyMem_Malloc(maxlen * sizeof(wchar_t));
+    wchar_t *buf = PyMem_New(wchar_t, maxlen);
```

Comment:
It never overflows. `maxlen` is at most `DWORD_MAX / 4`, and `sizeof(wchar_t)` is 2.